File: example/src/main/java/jp/kai/example/forcelayout/MainActivity.java
package jp.kai.example.forcelayout;

import android.app.Activity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;

import java.util.Arrays;
import java.util.HashMap;
import java.util.List;

import jp.kai.forcelayout.Forcelayout;
/**
 * Created by kai on 2016/09/03.
 */
public class MainActivity extends Activity {
    @Override protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

//        setContentView(R.layout.activity_main);
//        Forcelayout _bane = (Forcelayout) findViewById(R.id.sample_logview);

        final Forcelayout _bane =  new Forcelayout(getApplicationContext());

        Button button1 = new Button(this);
        button1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //set links
                List<String> links = Arrays.asList("neko-nyanko","neko1-neko2","neko5-neko8","neko-neko4","neko2-neko5");

                _bane.with(getApplicationContext()).linkStrength(0.09).gravity(0.04).distance(200).links(links);
            }
        });
        button1.setText("change links");

        Button button2 = new Button(this);
        button2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //set links
                List<String> links = Arrays.asList("neko-nyanko","neko1-neko2","neko5-neko8","neko-neko4","neko2-neko5");

                _bane.with(getApplicationContext()).linkStrength(0.1).gravity(0.3).distance(500);
            }
        });
        button2.setText("change linkStrength and gravity");


        LinearLayout layout = new LinearLayout(this);
        layout.setOrientation(LinearLayout.VERTICAL);
        layout.addView(button1);
        layout.addView(button2);
        layout.addView(_bane);
        setContentView(layout);



        //set nodes
        HashMap<String, Integer> nodes = new HashMap<>();

        nodes.put("neko",R.drawable.a);
        nodes.put("nyanko",R.drawable.b);
        nodes.put("hoge",R.drawable.c);
        nodes.put("neko1",R.drawable.d);
        nodes.put("neko2",R.drawable.d);
        nodes.put("neko3",R.drawable.d);
        nodes.put("neko4",R.drawable.d);
        nodes.put("neko5",R.drawable.d);
        nodes.put("neko6",R.drawable.d);
        nodes.put("neko7",R.drawable.e);
        nodes.put("neko8",R.drawable.e);
        nodes.put("neko9",R.drawable.f);
        nodes.put("neko10",R.drawable.f);
        nodes.put("neko11",R.drawable.g);
        nodes.put("neko12",R.drawable.g);

        //set links
        List<String> links = Arrays.asList("neko5-neko8","neko-neko4","neko1-neko2","neko2-neko3","neko3-neko1","neko5-neko10","neko3-neko11","neko3-neko12","neko3-nyanko","hoge-neko7");

        _bane.with(this)
                .nodesize(200)
                .linkStrength(0.09)
                .distance(200)
                .gravity(0.04)
                .nodes(nodes)
                .links(links);


    }


    @Override protected void onStart() {
        super.onStart();
    }

}

File: example/src/main/java/jp/kai/example/forcelayout/MainActivity.kt
package jp.kai.example.forcelayout

import android.app.Activity
import android.graphics.Color
import android.os.Bundle
import android.widget.SeekBar
import jp.kai.forcelayout.Forcelayout
import org.jetbrains.anko.button
import org.jetbrains.anko.seekBar
import org.jetbrains.anko.verticalLayout
import java.util.Arrays
import android.util.Pair

/**
 * Created by kai on 2016/09/03.
 * Usage for Kotlin
 */

class MainActivity : Activity() {
    private var isFlip: Boolean = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onStart() {
        super.onStart()

        val force = Forcelayout(this)

        example1(force)

        verticalLayout {
            button {
                text = "change examples"

                setOnClickListener {
                    if(isFlip){
                        example1(force)
                        isFlip = false
                    }else {
                        example2(force)
                        isFlip = true
                    }
                }
            }

            button {
                text = "change links"

                setOnClickListener {
                    /** set links */
                    val links = Arrays.asList("cat_a1-cat_a2", "cat_a2-cat_a3", "cat_a3-cat_d1", "cat_d1-cat_d2", "cat_d2-cat_e1")

                    force.with()
                            .friction(0.02)
                            .gravity(0.08)
                            .links(links)
                            .start()
                }
            }

            seekBar {
                max = 100

                setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
                    override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
                        force.with().gravity((progress.toFloat() / 1000).toDouble()).start()

                    }

                    override fun onStartTrackingTouch(seekBar: SeekBar) {

                    }

                    override fun onStopTrackingTouch(seekBar: SeekBar) {

                    }
                })
            }
            addView(force)
        }
    }

    private fun example1(force: Forcelayout){
        /** set nodes */
        val nodes: ArrayList<Pair<String, Int>> = ArrayList<Pair<String, Int>>()

        nodes.add(Pair("cat_a1", R.drawable.a))
        nodes.add(Pair("cat_a2", R.drawable.a))
        nodes.add(Pair("cat_a3", R.drawable.a))
        nodes.add(Pair("cat_b1", R.drawable.b))
        nodes.add(Pair("cat_c1", R.drawable.c))
        nodes.add(Pair("cat_d1", R.drawable.d))
        nodes.add(Pair("cat_d2", R.drawable.d))
        nodes.add(Pair("cat_e1", R.drawable.e))
        nodes.add(Pair("cat_f1", R.drawable.f))

        /** set links */
        val links: List<String> = Arrays.asList("cat_d1-cat_d2","cat_a1-cat_a3","cat_a2-cat_a3","cat_a1-cat_a2")

        force.with()
                .distance(350) /** distance between each nodes */
                .gravity(0.04) /** gravitation from center of view */
                .friction(0.02) /** value of gravitation between each nodes */
                .size(130) /** node width */
                .nodes(nodes) /** set nodes */
                .links(links) /** set links */
                .start() /** start animation */
    }

    private fun example2(force: Forcelayout){
        /** set nodes */
        val nodes: Array<String> = arrayOf( "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n","o","p","q","r")

        /** set links */
        val links: List<String> = Arrays.asList("a-b","c-d","d-g","r-o")

        force.node()
                .size(100)
                .style(Color.argb(100,200,30,50))

        force.link()
                .style(Color.argb(60,50,30,200), 5.0f)

        force.with()
                .distance(350) /** distance between each nodes */
                .gravity(0.04) /** gravitation from center of view */
                .friction(0.02) /** value of gravitation between each nodes */
                .nodes(nodes) /** set nodes */
                .links(links) /** set links */
                .start() /** start animation */
    }
}